src/EditorFeatures/Core.Cocoa/InlineRename/AbstractEditorInlineRenameService.cs

```diff
@@ -168,11 +168,11 @@ public async Task<IInlineRenameInfo> GetRenameInfoAsync(Document document, int p
                 else if (location.IsInSource)
                 {
                     var solution = document.Project.Solution;
-                    var sourceDocument = solution.GetDocument(location.SourceTree);
-                    if (sourceDocument == null)
+                    var sourceDocument = solution.GetRequiredDocument(location.SourceTree);
+
+                    if (sourceDocument is SourceGeneratedDocument)
```

Comment:
i disliek that this is a type check, rather than some sort of property on Document.   Also, how does GetRequiredDocument work here?  is it just known that thsi must succeed without asynchrony in order for us to map back from a tree to a doc?

Reply:
Yeah, this is the case where we don't need asynchrony: the only way you could have ended up with a syntax tree was if the generators already ran.

Regarding the type check, https://github.com/dotnet/roslyn/issues/50297 is tracking looking at whether we should update our kind property or not. There's arguments both ways for what the consumer expects, but for now this the option.